multiversx_sdk_cli/contracts.py

```diff
@@ -232,6 +216,42 @@ def _interpret_as_number_if_safely(as_hex: str) -> Optional[int]:
         return None
 
 
+def prepare_execute_transaction_data(function: str, arguments: List[Any]) -> TransactionPayload:
+    tx_data = function
+
+    for arg in arguments:
+        tx_data += f"@{_prepare_argument(arg)}"
+
+    return TransactionPayload.from_str(tx_data)
+
+
+def prepare_args_for_factory(arguments: List[str]) -> List[Any]:
+    args: List[Any] = []
+
+    for arg in arguments:
+        if arg.startswith(HEX_PREFIX):
+            args.append(hex_to_bytes(arg))
+        elif arg.isnumeric():
+            args.append(int(arg))
+        elif arg.startswith(DEFAULT_HRP):
+            args.append(Address.new_from_bech32(arg))
+        elif arg.lower() == FALSE_STR_LOWER:
+            args.append(False)
+        elif arg.lower() == TRUE_STR_LOWER:
+            args.append(True)
+        elif arg.startswith(STR_PREFIX):
+            args.append(arg[len(STR_PREFIX):])
+
+    return args
+
+
+def hex_to_bytes(arg: str):
+    argument = arg[len(HEX_PREFIX):]
+    argument = argument.upper()
+    argument = ensure_even_length(argument)
+    return bytes.fromhex(argument)
+
+
```

Comment:
Below, `_prepare_argument` and `_to_hex` - since this PR, they are only needed for the queries and for some stake operations etc., correct? Perhaps add a comment about this? Maybe we will have a replacement for this in the future.

Reply:
Added a short comment.